### PR TITLE
Fix for bug 1109 & 459: Nested groups does not folding in

### PR DESCRIPTION
**Pull request by ktayal-tech**
Fixed bug 1109: Nested groups does not folding in #1109. 
Fixed bug 459: Deeply nested subgroups can cause subgroups to become displaced, appearing at the top of the timeline #459

### CODE REVIEW

File: lib/timeline/component/ItemSet.js [hunk 1]
```diff
@@ -1946,13 +1946,34 @@ class ItemSet extends Component {
...
+    if(nestingGroup.showNested){
+      var showNestedGroups=groupsData.get(nestingGroup.nestedGroups);
+        let i=0;
+        while (i<showNestedGroups.length){
+          let group=showNestedGroups[i];
+          if(group.nestedGroups && group.nestedGroups.length>0 && (group.showNested == undefined || group.showNested == true)){
+            showNestedGroups.push(...groupsData.get(group.nestedGroups));
+          }
+          i=i+1;
+        }
```
Comment (mojoaxel) [inline lib/timeline/component/ItemSet.js:1959]: Should be a for loop.

File: lib/timeline/component/ItemSet.js [hunk 2]
```diff
@@ -1946,13 +1946,34 @@ class ItemSet extends Component {
...
+    }
+    else{
+     nestedGroups = groupsData.get(fullNestedGroups).map(function (nestedGroup) {
         if (nestedGroup.visible == undefined) {
           nestedGroup.visible = true;
         }
         nestedGroup.visible = !!nestingGroup.showNested;
         return nestedGroup;
-      });
+      });}
```
Comment (mojoaxel) [inline lib/timeline/component/ItemSet.js:1976]: please fix the code formatting to fit to the rest of the code-base.

**Review by mojoaxel — CHANGES_REQUESTED** [review]
@ktayal-tech Thank you very much!

The code works but needs a "face-lift" to fit the code-style of the rest of the file.
Consider to use a tool like [prettier](https://prettier.io/) but make sure you only change the lines of code that you added. Thanks!

File: lib/timeline/component/ItemSet.js (hunk 1 above)
Comment (ktayal-tech) [reply lib/timeline/component/ItemSet.js]: Thank you @mojoaxel, same has been done.

File: lib/timeline/component/ItemSet.js (hunk 2 above)
Comment (ktayal-tech) [reply lib/timeline/component/ItemSet.js]: Done using 'prettier'.

**Review by mojoaxel — APPROVED** [review]
Thanks! Looks good to me!

**Pull request merged**